spring-integration-core/src/test/java/org/springframework/integration/aop/MethodAnnotationPublisherMetadataSourceTests.java

```diff
@@ -83,6 +84,13 @@ public void payloadAndHeaders() {
 		assertEquals("#args[2]", headerMap.get("bar").getExpressionString());
 	}
 
+	@Test
+	public void expressionsAreConcurrentHashMap() {
+		assertTrue("Expressions should be concurrent to allow startup", getField(source, "channels") instanceof ConcurrentHashMap);
```

Comment:
There is `assertThat()` and `Matchers.instanceOf()` for better failure report if that.

Reply:
Great suggestion. Updated.